### PR TITLE
Fixes #170

### LINKED ISSUE #170: Update on Frontend Introjs Tour Popup Part 2
**Original Issue** https://github.com/oss-slu/lrda_website/issues/138

**Description**

**As** a new user,
**I want** my on-boarding tour to have good UI Design,
**So that** I can easily navigate and use the map functionalities and refer back to the tour through the help section if needed.

**Tasks**
Install Intro.js:
- [x] Create a test case for your code
- [ ] Adjust the map search highlight on IntroJs Popup
       - make it smaller
       - move button a little to the left
       - inspect overall design of the popup window and make it more atheistic
- [ ] adjust popup tour so it only shows up for user one time
       - a skip button or do not show button will stop the tour from showing
       - if a user went through the tour once it will not show up again
**Acceptance Criteria**
- [ ] Create at least one test case for you code
- [ ] Code is well commented
- [ ] Only commit changes that you physically typed.
- [ ] Code should pass the test case before merge

### PR DESCRIPTION
**Pull request by amychen108**
**1. PR Reference**
Fixes #170 
**2. What was changed:**
Updating the introjs to enhance visual appeal by adjusting the popup's design for the search bar specifically, adding functionality to ensure the tour shows only once per user, and making sure all test cases still pass after changes
**3. Why it was changed:**
This update improves user experience by refining the search bars' popup aesthetics and functionality, making the tour both visually appealing and non-intrusive. This update also helps avoid redundant interruptions by making the tour only appears once per user.
**4. How it was changed:**
**UI update:**
The search bar reference was moved to a different css container so that the intro popup will point to a better reference for the search bar
**Tour Display Logic:**
The tour shows up only once per user by implementing a “do not show again” button.
If a user has already completed or skipped the tour, it will not appear again in future sessions.

Testing:
Writing test case for making sure the new css container shows up properly. 
Making sure all previous tests still pass. (regression testing)

### CODE REVIEW

File: app/__tests__/search_bar_ref.test.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  return (
+    <input
+      type="text"
+      placeholder="Search"
+      ref={searchBarRef}
+    />
+  );
+}
+
+test('search bar ref is assigned to the correct search bar element', () => {
```
Comment (jwang-101) [inline app/__tests__/search_bar_ref.test.tsx:19]: test case looks good, properly references the search bar with jest mock

File: app/lib/pages/map/page.tsx [hunk 2]
```diff
@@ -645,11 +645,11 @@ const Page = () => {
             }}
           >
             <div className="absolute flex flex-row mt-3 w-full h-10 justify-between z-10">
-             {/* *** Attach ref to Search Bar Conatiner *** */}
               <div 
-                className="flex flex-row w-[30vw] left-0 z-10 m-5 align-center items-center"
+                className="flex flex-row w-[30vw] left-0 z-10 m-5 align-center items-center">
```
Comment (jwang-101) [inline app/lib/pages/map/page.tsx:649]: this seems like a better placement for the pop up size on the search bar

**Review by jwang-101 — COMMENTED** [review]
the change looks aesthetically better and the test case looks good

**Review by BaloneyBoy97 — APPROVED** [review]
great work Amy!! the new css change looks much cleaner! PR APPROVED!!

**Pull request merged**